IDCBrowser/IDCHandlerModule.py

```diff
@@ -0,0 +1,407 @@
+Name=IDC Browser
+Exec={python_path} {python_script_path} %u
+Type=Application
+Terminal=false
+MimeType=x-scheme-handler/idcbrowser;
+"""
+                )
+            # Update MIME database
+
+            os.system("update-desktop-database ~/.local/share/applications/")
+            os.system("xdg-mime default idcbrowser.desktop x-scheme-handler/idcbrowser")
+        elif platform.system() == "Windows":
+
+            # Get the directory of the current Python executable
+
+            python_dir = os.path.dirname(sys.executable)
+
+            # Construct the path to PythonSlicer.exe in the same directory
+
+            python_path = os.path.join(python_dir, "PythonSlicer.exe")
+
+            current_dir = os.path.dirname(os.path.realpath(__file__))
+            python_script_path = os.path.join(current_dir, "Resources", "resolver.py")
+
+            # Register IDC Browser URL protocol in Windows Registry
+
+            import winreg as reg
+
+            try:
+                reg.CreateKey(reg.HKEY_CURRENT_USER, r"Software\Classes\idcbrowser")
+                with reg.OpenKey(
+                    reg.HKEY_CURRENT_USER,
+                    r"Software\Classes\idcbrowser",
+                    0,
+                    reg.KEY_WRITE,
+                ) as key:
+                    reg.SetValue(key, None, reg.REG_SZ, "URL:IDC Browser Protocol")
+                    reg.SetValueEx(key, "URL Protocol", 0, reg.REG_SZ, "")
+                reg.CreateKey(
+                    reg.HKEY_CURRENT_USER,
+                    r"Software\Classes\idcbrowser\shell\open\command",
+                )
+                with reg.OpenKey(
+                    reg.HKEY_CURRENT_USER,
+                    r"Software\Classes\idcbrowser\shell\open\command",
+                    0,
+                    reg.KEY_WRITE,
+                ) as key:
+                    reg.SetValue(
+                        key,
+                        None,
+                        reg.REG_SZ,
+                        f'"{python_path}" "{python_script_path}" "%1"',
+                    )
+                print("IDC Browser URL protocol has been registered on Windows.")
+            except Exception as e:
+                print(f"Failed to register IDC Browser URL protocol on Windows: {e}")
+        elif platform.system() == "Darwin":
+            slicer_exec_dir = os.path.dirname(sys.executable)
+            parent_dir = os.path.dirname(slicer_exec_dir)
+
+            # Now, you can construct the path to PythonSlicer
+
+            python_path = shlex.quote(os.path.join(parent_dir, "bin", "PythonSlicer"))
+
+            current_dir = os.path.dirname(os.path.realpath(__file__))
+            python_script_path = shlex.quote(
+                os.path.join(current_dir, "Resources", "resolver.py")
+            )
+
+            def check_macos_slicer_protocol_registration():
+                plist_path = os.path.expanduser(
+                    "/Applications/slicer-app.app/Contents/Info.plist"
```

Comment:
I think you would want to install this in `~/Applications` rather than in the system-wide `/Applications` since the handler is installed in the user's directory.  (Assuming it works in the `~/Applications` directory).